workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/WorkflowLayout.kt

```diff
@@ -47,25 +51,89 @@ public class WorkflowLayout(
   private var restoredChildState: SparseArray<Parcelable>? = null
 
   /**
-   * While this view is attached to a window, subscribes to [renderings] and display its values.
+   * Calls [WorkflowViewStub.show] on the [WorkflowViewStub] that is the only
+   * child of this view.
    *
-   * To configure the [ViewEnvironment], use [EnvironmentScreen] as your rendering type.
-   * For example, to customize the UI via [ViewRegistry] entries:
+   * It's more common for a `Workflow`-based `Activity` or `Fragment` to use
+   * [take] than to call this method directly. It is exposed to allow clients to
+   * make their own choices about how exactly to consume a stream of renderings.
+   */
+  public fun show(rootScreen: Screen) {
+    showing.show(rootScreen, rootScreen.withEnvironment().viewEnvironment)
+    restoredChildState?.let { restoredState ->
+      restoredChildState = null
+      showing.actual.restoreHierarchyState(restoredState)
+    }
+  }
+
+  /**
+   * This is the most common way to bootstrap a [Workflow][com.squareup.workflow1.Workflow]
+   * driven UI. Collects [renderings] and calls [show] with each one.
    *
-   *     val registry = ViewRegistry(MuchBetterViewForFooScreen)
-   *     val renderings: Flow<EnvironmentScreen<*>> = renderWorkflowIn(...).map {
-   *       it.withRegistry(registry)
-   *     }
-   *     workflowLayout.take(renderings)
+   * @param [lifecycle] the lifecycle that defines when and how this view should be updated.
+   * Typically this comes from `ComponentActivity.lifecycle` or  `Fragment.lifecycle`.
    */
-  public fun take(renderings: Flow<Screen>) {
-    takeWhileAttached(renderings.map { it.withEnvironment() }) { show(it) }
+  public fun take(
+    lifecycle: Lifecycle,
+    renderings: Flow<Screen>
+  ) {
+    // Just like https://medium.com/androiddevelopers/a-safer-way-to-collect-flows-from-android-uis-23080b1f8bda
+    lifecycle.coroutineScope.launch {
+      lifecycle.repeatOnLifecycle(Lifecycle.State.STARTED) {
+        renderings.collect { show(it.withEnvironment()) }
+      }
+    }
+  }
+
+  @Deprecated(
+    "Use show",
+    ReplaceWith(
+      "show(asScreen(newRendering).withEnvironment(environment))",
+      "com.squareup.workflow1.ui.container.withEnvironment",
+      "com.squareup.workflow1.ui.asScreen"
+    )
+  )
+  public fun update(
```

Comment:
Maybe should rename this to `show` on `main` before the next release, so that there will be one less bit of swirl when the overhaul lands.

Reply:
Too late, I forgot that we've already released it that way.